src/generator.cs

```diff
@@ -1401,19 +1540,24 @@ public string MarshalParameter (MethodInfo mi, ParameterInfo pi, bool null_allow
 		throw new BindingException (1002, true, "Unknown kind {0} in method '{1}.{2}'", pi, mi.DeclaringType.FullName, mi.Name.GetSafeParamName ());
 	}
 
-	public bool ParameterNeedsNullCheck (ParameterInfo pi, MethodInfo mi)
+	public bool ParameterNeedsNullCheck (ParameterInfo pi, MethodInfo mi, PropertyInfo propInfo = null)
 	{
 		if (pi.ParameterType.IsByRef)
 			return false;
 
 		if (HasAttribute (pi, typeof (NullAllowedAttribute)))
 			return false;
 
-		if (mi.IsSpecialName && mi.Name.StartsWith ("set_", StringComparison.Ordinal)){
+		if (IsSetter (mi)) {
 			if (HasAttribute (mi, typeof (NullAllowedAttribute))){
 				return false;
 			}
 		}
+
+		var bindAsAtt = GetBindAsAttribute (pi) ?? GetBindAsAttribute (propInfo);
+		if (bindAsAtt != null)
+			return bindAsAtt.IsNullable;
```

Comment:
Is this correct for reference types as well (which can be null, but IsNullable will be false)?